### PR TITLE
Complete the skeleton for draft-00

### PR DESCRIPTION
**Pull request by cjpatton**
The bits that we should fill out before cutting the draft are marked with "XXX". Ignore "TODO" for now: my intention is that we would fill these out once the spec is more baked.

### CODE REVIEW

File: draft-chen-cfrg-vdaf-pine.md [hunk 1]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
-  the probability of false positives negligible.
-* The random vectors in wraparound check need to be generated in a
-  non-interactive way and agreed by both the Client and Aggregators.
-  The randomness generation technique is very similar to the technique
-  used for joint randomness in Prio3 (Fiat-Shamir heuristic), and we
-  refer to it as "wraparound joint randomness".
-
-# An FLP for Pine {#flp}
+Rather than prevent wraparounds from occuring, PINE uses a probabilistic test
+to detect whether a wraparound has occured. This test works as follows. A
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:226]: I don't think Aggregators can ever prevent wraparounds from occurring? How about:

```suggestion
In order to detect whether a wraparound has occurred, PINE uses a probabilistic test,
which works as follows: A
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 2]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
+random vector over the field is generated (via a procedure described in
+{{vdaf}}) where each entry is equal to `1`, `0`, or `q-1`, each with a
+particular probability: to test for wraparound, compute the dot product of this
+vector and the gradient and check if the result in a specific range. The range
+is determined by parameters in {{pine-user-param}}.
+
+If the norm wraps around the field modulus, then the dot product is likely to
+be large. In fact, {{ROCT23}} show that this test correctly detects warparounds
+with probability `1/2`. To decrease the false negative probability (that is,
+the probability of missclassifying an invalid gradient as valid), we simply
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:236]:
```suggestion
the probability of misclassifying an invalid gradient as valid), we simply
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 3]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
+repeat this test a number of times, each time with a vector sampled from the
+same distribution.
+
+However, {{ROCT23}} also show that each wraparound test has a non-zero false
+positive probability (the probability of missclassifying a valid gradient as
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:241]:
```suggestion
positive probability (the probability of misclassifying a valid gradient as
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 4]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
+dot product with a known vector is in a particular range. [CP: We need a more
+intuitive explanation of the information that's leaked.] The parameters of PINE
+are chosen carefully in order to ensure this leakage is negligible.
+
+# The PINE Proof System {#flp}
+
+This section specifies a randomized encoding of gradients and FLP circuit
+({{Section 7.3 of !VDAF}}) for checking that (1) computes the gradient's
+squared L2-norm and checks that it falls in the desired range and (2) checking
+that the squared L2-norm does not wrap around the field modulus.
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:253]:
```suggestion
({{Section 7.3 of !VDAF}}) for checking that (1) the gradient's
squared L2-norm falls in the desired range and (2) the squared L2-norm does not wrap around the field modulus.
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 5]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
+
+The encoding algorithm takes as input the gradient and an XOF seed used to
+derive the random vectors for the wraparoundt tests. The seed must be known
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:256]:
```suggestion
derive the random vectors for the wraparound tests. The seed must be known
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 6]
```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
...
+
+# Security Considerations
+
+Our security considerations for PINE are the same as those for Prio3 described
+in {{Section 9 of !VDAF}}.
+
+> XXX Given that we can tune the parameters such that the ZK error is
+> negligible, I don't think this needs to be mentioned here.
+>
+> Is there anything else worth mentioning?
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:360]: The other difference from Prio3 is there are two Fiat-Shamir derivation, so the robustness needs to account for the random oracle attack from both wraparound joint randomness and verification joint randomness?

Comment (cjpatton) [reply draft-chen-cfrg-vdaf-pine.md:360]: Our goal will be to tune the parameters such that the user doesn't have to worry about this.
